src/v/kafka/server/handlers/fetch.cc

```diff
@@ -261,36 +262,42 @@ static void fill_fetch_responses(
         auto& resp_it = responses[idx];
 
         // error case
-        if (!res.has_data()) {
+        if (unlikely(res.error != error_code::none)) {
             resp_it.set(
               make_partition_response_error(res.partition, res.error));
             continue;
         }
+
+        model::ntp ntp(
+          model::kafka_namespace,
+          resp_it->partition->name,
+          resp_it->partition_response->partition_index);
+        /**
+         * Cache fetch metadata
+         */
+        octx.rctx.get_fetch_metadata_cache().insert_or_assign(
+          std::move(ntp),
+          res.start_offset,
+          res.high_watermark,
+          res.last_stable_offset);
+        /**
+         * Over response budget, we will just waste this read, it will cause
+         * data to be stored in the cache so next read is fast
+         */
         fetch_response::partition_response resp;
         resp.partition_index = res.partition;
         resp.error_code = error_code::none;
         resp.log_start_offset = res.start_offset;
         resp.high_watermark = res.high_watermark;
         resp.last_stable_offset = res.last_stable_offset;
 
-        /**
-         * set aborted transactions if present
-         */
-        if (!res.aborted_transactions.empty()) {
```

Comment:
what happened with aborted transactions? we used to pass them to the response object there where do you pass them now?

Reply:
you are right, I missed that when refactoring this part. great catch